src/utils/useChangedDepsLogger.ts

```diff
@@ -0,0 +1,18 @@
+import {useRef, MutableRefObject} from 'react'
+
+export const useChangedDepsLogger = deps => {
```

Comment:
would you mind to providing an example of how to use this in a comment? Looks useful.